unstract/connectors/src/unstract/connectors/filesystems/sftp/sftp.py

```diff
@@ -79,3 +98,45 @@ def test_credentials(self) -> bool:
                 "please check the connection settings."
             )
         return True
+
+    def create_dir_if_not_exists(self, input_dir: str) -> None:
+        """Method to create dir of a connector if not exists.
+
+        Args:
+            input_dir (str): input directory of source connector
+        """
+        fs_fsspec = self.get_fsspec_fs()
+        try:
+            is_dir = fs_fsspec.isdir(input_dir)
+            if not is_dir:
+                fs_fsspec.mkdir(input_dir)
+        except PermissionError as e:
+            self.raise_permission_error(input_dir=input_dir, error=e)
+
+    def upload_file_to_storage(self, source_path: str, destination_path: str) -> None:
+        """Method to upload filepath from tool to sftp connector directory.
+
+        Args:
+            source_path (str): local path of file to be uploaded, coming from tool
+            destination_path (str): target path in the storage where the file will be
+            uploaded
+        """
+        try:
+            normalized_path = os.path.normpath(destination_path)
+            fs = self.get_fsspec_fs()
+            with open(source_path, "rb") as source_file:
+                fs.write_bytes(normalized_path, source_file.read())
+        except PermissionError as e:
+            self.raise_permission_error(input_dir=normalized_path, error=e)
+
+    def raise_permission_error(
```

Comment:
NIT: I feel that this doesn't warrant another function - you could try to move this `user_message` to the separate `PermissionDeniedError` class instead

Reply:
@chandrasekharan-zipstack this user_msg is specific to sftp permission denied error, let us not move that to `PermissionDeinedError`

Reply:
This is okay for the time being however, I'd still recommend to not over-complicate. Messages can be made generic as well or arguments can be passed to customize. Let's refactor when a need arises